api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -146,20 +148,24 @@ export class ScenariosController {
   // TODO add Validations
   @ApiConsumesShapefile()
   @Post(':id/planning-unit-shapefile')
-  //@UseInterceptors(FileInterceptor('file', uploadOptions))
+  @UseInterceptors(FileInterceptor('file', uploadOptions))
   async uploadLockInShapeFile(
-    @Param('id') scenarioId: string,
-    @Req() request: Request,
-    @Res() response: Response,
+    @Param('id', ParseUUIDPipe) scenarioId: string,
+    @UploadedFile() file: Express.Multer.File,
   ): Promise<ShapefileGeoJSONResponseDTO> {
     await this.service.getById(scenarioId);
-    const proxyServiceResponse = await this.proxyService.proxyUploadShapeFile(
-      request,
-      response,
-    );
-    return proxyServiceResponse;
+    const { data } = await this.httpService
+      .post(
+        `${this.geoprocessingUrl}${apiGlobalPrefixes.v1}/planning-units/planning-unit-shapefile`,
+        file,
+        {
+          headers: { 'Content-Type': 'application/json' },
+          validateStatus: (status) => status <= 499,
```

Comment:
Are we really fine with the 4xx range? While flicking through the code I do find it surprising that we resolve the promise for 4xx responses from downstream - maybe a quick tiny comment to explain why could help here.

Reply:
@hotzevzl I added this because **axios** (wrapped by HttpService) rejects the promise if status > 399, so if we return a response above that in geoprocessing (because shapefile is not validated), axios wraps that message and the API consumer does not get a 'Invalid Shapefile' response

We could add a trycatch block here but this seemed the best option to me.

Reply:
ok, I understand, indeed it makes sense as we're only proxying. thanks!